parser/tools/generate_ast.cc

```diff
@@ -204,6 +204,8 @@ NodeDef nodes[] = {
         "defined?",
         vector<FieldDef>({{"value", FieldType::Node}}),
     },
+    // def name, instance method def
+    {"DefnHead", "defnhead", vector<FieldDef>({{"name", FieldType::Name}})},
```

Comment:
This used to be represented by a node list. Now we create an explicit node so that it can be used to allocate a `node_with_context`.